eval.ml

```diff
@@ -94,11 +95,19 @@ let value_eq (v1 : value) (v2 : value) =
   | Int n1, Int n2 -> n1 = n2
   | Int _, _ | _, Int _ -> false
   | Fun_ref f1, Fun_ref f2 -> f1 = f2
+  | Fun_ref _, _ | _, Fun_ref _ -> false
+  | Array vs1, Array vs2 ->
```

Comment:
I'm fine with leaving it as is -- maybe I was just reading it too quickly and thought some patterns wer redundant. And if someone forgets a case, the compiler should complain.